### PR TITLE
Refactor `Cleanup`.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Start refactoring `Cleanup`.

### CODE REVIEW

File: Library/Homebrew/cleanup.rb [hunk 1]
```diff
@@ -1,6 +1,71 @@
 require "utils/bottles"
 require "formula"
 
+module CleanupRefinement
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cleanup.rb]: Is this replacing monkeypatching or just makes the code below a bit nicer?

File: Library/Homebrew/cleanup.rb [hunk 2]
```diff
@@ -46,22 +111,17 @@ def cleanup_keg(keg)
     def cleanup_logs
       return unless HOMEBREW_LOGS.directory?
       HOMEBREW_LOGS.subdirs.each do |dir|
-        cleanup_path(dir) { dir.rmtree } if prune?(dir, days_default: 14)
+        cleanup_path(dir) { dir.rmtree } if dir.prune?(ARGV.value("prune")&.to_i || 14)
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cleanup.rb]: I think the `14` should become a constant as it's no longer documented by the `days_default` now

File: Library/Homebrew/cleanup.rb [hunk 3]
```diff
@@ -1,6 +1,71 @@
...
+
+    def nested_cache?
+      directory? && ["glide_home", "java_cache", "npm_cache"].include?(basename.to_s)
+    end
+
+    def prune?(days)
+      return false unless days
+      return true if days.zero?
+
+      # TODO: Replace with ActiveSupport's `.days.ago`.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/cleanup.rb]: Do you want this to block the PR?

**Comment by MikeMcQuaid** [comment]
@reitermarkus Nice work here. Just to elaborate a bit on some of my intentions here so you might be able to do some stuff to help: I was hoping to make `brew upgrade --cleanup` (and `HOMEBREW_UPGRADE_CLEANUP`) cleanup more stuff related to a formula e.g. the `cache` and perhaps prune old symlinks (`brew prune`) too. On top of this I was hoping eventually to get to the point where `brew linkage` data (now that it's cached) could be used to run `brew cleanup` automatically on every upgrade/install (guarded by `HOMEBREW_UPGRADE_CLEANUP` to begin with) if a keg isn't being linked to by any other active keg.

File: Library/Homebrew/cleanup.rb (hunk 1 above)
Comment (reitermarkus) [reply Library/Homebrew/cleanup.rb]: Just to make the actual cleanup logic easier to read.

File: Library/Homebrew/cleanup.rb (hunk 3 above)
Comment (reitermarkus) [reply Library/Homebrew/cleanup.rb]: No, this is just so we don't forget to use it when we have it.

**Pull request merged**